Stripped down config fails to compile - ESPmDNS.h: No such file

<!-- Thanks for reporting a bug for this project. READ THIS FIRST:

- Provide as many details as possible. Simply saying "X gives bug" or "X gives error" is not enough!
- Paste logs, configuration sample and code into the backticks (```).
- Read through the template carefully and fill out all missing details.
- Please also search for similar issues in this issue tracker first and read through the ESPHome FAQ.

DO NOT DELETE ANY TEXT from this template! Otherwise the issue may be closed without a comment.
-->

**Operating environment/Installation (Hass.io/Docker/pip/etc.):**
pip with esphome v1.11.0

**ESP (ESP32/ESP8266, Board/Sonoff):**
esp32
**Affected component:**
**Description of problem:**
Failure to compile. I get this error
```
.piolibdeps/esphome-core/src/esphome/util.cpp:9:23: fatal error: ESPmDNS.h: No such file or directory
```


If i add `ota:` then it does compile

**Problem-relevant YAML-configuration entries:**
```yaml
esphome:                                                                                                                                                                                                                                                                                  
  name: hottub                                                                                                                                                                                                                                                                            
  platform: ESP32                                                                                                                                                                                                                                                                         
  board: nodemcu-32s                                                                                                                                                                                                                                                                      
logger:                                                                                                                                                                                                                                                                                   
sensor:                                                                                                                                                                                                                                                                                   
  - platform: dht                                                                                                                                                                                                                                                                         
    pin:                                                                                                                                                                                                                                                                                  
      number: GPIO22                                                                                                                                                                                                                                                                      
      mode: INPUT_PULLUP                                                                                                                                                                                                                                                                  
    model: DHT22                                                                                                                                                                                                                                                                          
    temperature:                                                                                                                                                                                                                                                                          
      name: "Outside Temperature"                                                                                                                                                                                                                                                         
    humidity:                                                                                                                                                                                                                                                                             
      name: "Outside Humidity"                                                                                                                                                                                                                                                            
    update_interval: 1s           

```

--- a/esphome/components/ota.py
+++ b/esphome/components/ota.py
@@ -51,7 +51,7 @@ REQUIRED_BUILD_FLAGS = '-DUSE_NEW_OTA'
 
 def lib_deps(config):
     if CORE.is_esp32:
-        return ['Update', 'ESPmDNS']
+        return ['Update']
     if CORE.is_esp8266:
-        return ['Hash', 'ESP8266mDNS']
+        return ['Hash']
     raise NotImplementedError

--- a/esphome/writer.py
+++ b/esphome/writer.py
@@ -288,8 +288,10 @@ def gather_lib_deps():
         if CORE.arduino_version in (ARDUINO_VERSION_ESP32_DEV, ARDUINO_VERSION_ESP32_1_0_1):
             lib_deps.add('AsyncTCP@1.0.3')
             lib_deps.discard('AsyncTCP@1.0.1')
+        lib_deps.add('ESPmDNS')
     elif CORE.is_esp8266:
         lib_deps.add('ESPAsyncTCP@1.1.3')
+        lib_deps.add('ESP8266mDNS')
     # avoid changing build flags order
     lib_deps_l = list(lib_deps)
     lib_deps_l.sort()
@@ -393,7 +395,6 @@ def get_ini_content():
         data['lib_ldf_mode'] = 'chain'
         REMOVABLE_LIBRARIES = [
             'ArduinoOTA',
-            'ESPmDNS',
             'Update',
             'Wire',
             'FastLED',

